docs/tutorials/plugin-provider.md

```diff
@@ -0,0 +1,15 @@
+# Plugin provider
+
+The Plugin provider is a provider that allows DNS providers to integrate with ExternalDNS via an HTTP interface. The Plugin provider implements the Provider interface but instead of implementing code specific to a provider, it implements and HTTP client that sends request to an HTTP API. The idea behind this is that providers can be implemented in a separate program that exposes an HTTP API that the Plugin provider can interact with. The ideal setup for providers is to run as sidecars of the ExternalDNS container, but this is not strictly a requirement.
```

Comment:
```suggestion
The Plugin provider is a provider that allows DNS providers to integrate with ExternalDNS via an HTTP interface. The Plugin provider implements the Provider interface but instead of implementing code specific to a provider, it implements an HTTP client that sends request to an HTTP API. The idea behind this is that providers can be implemented in a separate program that exposes an HTTP API that the Plugin provider can interact with. The ideal setup for providers is to run as sidecars of the ExternalDNS container and listen on localhost only, but this is not strictly a requirement.
Be aware that the Plugin provider client does neither support authentication nor authorization, so exposing the plain Plugin provider over the network can put the integrity of your DNS records at risk.  
```

I feel like there should be a stronger message here as this could become a security issue quickly, if the user is holding it wrong.

Reply:
Yeah localhost is good because we do not need to think about authnz nor secret rotation.

Reply:
You are right, this could become problematic, let me add it to the text.